### PR TITLE
Added API to read and trim a stream.

### PR DESCRIPTION
**Pull request by MeirShpilraien**
Taken from: https://github.com/RedisLabsModules/redismodule-rs/pull/229
Added API for stream integration.

### CODE REVIEW

File: src/stream.rs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+pub struct StreamIterator<'key> {
+    key: &'key RedisKey,
+}
+
+impl<'key> StreamIterator<'key> {
+    pub(crate) fn new(
+        key: &RedisKey,
+        mut from: Option<raw::RedisModuleStreamID>,
+        mut to: Option<raw::RedisModuleStreamID>,
+        exclusive: bool,
```
Comment (gkorland) [inline src/stream.rs:22]: @MeirShpilraien what about REDISMODULE_STREAM_ITERATOR_REVERSE?

File: src/stream.rs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                flags,
+                from.as_mut().map_or(ptr::null_mut(), |v| v),
+                to.as_mut().map_or(ptr::null_mut(), |v| v),
+            )
+        };
+        if res != raw::REDISMODULE_OK as i32 {
+            Err(RedisError::Str("Failed creating stream iterator"))
+        } else {
+            Ok(StreamIterator { key })
+        }
```
Comment (gkorland) [inline src/stream.rs:41]:
```suggestion
if Status::Ok == res.into() {
            Ok(StreamIterator { key })
        } else {
            Err(RedisError::Str("Failed creating stream iterator"))
        }
```

File: src/stream.rs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        let mut num_fields: c_long = 0;
+        let mut field_name: *mut raw::RedisModuleString = ptr::null_mut();
+        let mut field_val: *mut raw::RedisModuleString = ptr::null_mut();
+        if unsafe {
+            raw::RedisModule_StreamIteratorNextID.unwrap()(
+                self.key.key_inner,
+                &mut id,
+                &mut num_fields,
+            )
+        } != raw::REDISMODULE_OK as i32
```
Comment (gkorland) [inline src/stream.rs:59]:
```suggestion
        if Status::Ok
            != unsafe {
                raw::RedisModule_StreamIteratorNextID.unwrap()(
                    self.key.key_inner,
                    &mut id,
                    &mut num_fields,
                )
            }
            .into()
```

File: src/stream.rs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            return None;
+        }
+        let mut fields = Vec::new();
+        while unsafe {
+            raw::RedisModule_StreamIteratorNextField.unwrap()(
+                self.key.key_inner,
+                &mut field_name,
+                &mut field_val,
+            )
+        } == raw::REDISMODULE_OK as i32
```
Comment (gkorland) [inline src/stream.rs:70]:
```suggestion
        while Status::Ok
            == unsafe {
                raw::RedisModule_StreamIteratorNextField.unwrap()(
                    self.key.key_inner,
                    &mut field_name,
                    &mut field_val,
                )
                .into()
            }
```

**Review by gkorland — CHANGES_REQUESTED** [review]

File: src/stream.rs (hunk 1 above)
Comment (MeirShpilraien) [reply src/stream.rs:22]: Good point, will add it.

Comment (MeirShpilraien) [reply src/stream.rs:23]: Done

**Review by gkorland — APPROVED** [review]

**Pull request merged**